main.go

```diff
@@ -183,8 +186,8 @@ func main() {
 
 	router := route.New()
 
-	RegisterWeb(router)
-	api.Register(router.WithPrefix("/api"))
+	RegisterWeb(router.WithPrefix(amURL.Path))
+	api.Register(router.WithPrefix(amURL.Path + "/api"))
```

Comment:
path.Join is fine in general, but can only be applied to paths, not full URLs. Otherwise it'll get rid of duplicated slashes and do all kinds of other things you might not want.